trl/trainer/sft_trainer.py

```diff
@@ -249,6 +249,13 @@ def make_inputs_require_grad(module, input, output):
                     "You passed `packing=False` to the SFTTrainer, but you didn't pass a `dataset_text_field` or `formatting_func` argument."
                 )
 
+            if args is not None and not args.remove_unused_columns and data_collator is None:
+                warnings.warn(
+                    "You passed `remove_unused_columns=False` on a non-packed dataset while using `DataCollatorForLanguageModeling`. This might create some issues with the default collator. If you want to "
+                    "inspect dataset other columns, you can subclass `DataCollatorForLanguageModeling` and create your own data collator in order to inspect the unused dataset columns."
+                )
+                args.remove_unused_columns = True
```

Comment:
I am not sure this is the right approach here, potentially overwriting a user choice (`args.remove_unused_columns = False`) or warning when no warning is needed (if there are no extra columns).

Reply:
Maybe we can check if extra columns exist and suggesting setting removed unused columns to `True` if that's the case?